samcli/commands/deploy/auth_utils.py

```diff
@@ -101,6 +101,13 @@ def _auth_id(resources_dict, event_properties, identifier):
     """
     resource_name = event_properties.get(identifier, "")
     api_resource = resources_dict.get(resource_name, {})
+
+    # Auth does not apply to ApiGateway::RestApi resources so return true and continue
+    if api_resource and (
+        api_resource["Type"] == "AWS::ApiGateway::RestApi" or api_resource["Type"] == "AWS::ApiGatewayV2::Api"
```

Comment:
Let's use `.get()` instead to avoid accidental `AttributeErrors` in the case of a malformed template.
```suggestion
        api_resource.get("Type", "") == "AWS::ApiGateway::RestApi" or api_resource.get("Type", "") == "AWS::ApiGatewayV2::Api"
```